lambda_functions/transform_cloudwatch_lambda.py

```diff
@@ -0,0 +1,144 @@
+                output_data = (
+                    "\n".join([json.dumps(log) for log in processed_logs]) + "\n"
+                )
+
+                # Just base64 encode for Firehose transport (no gzip)
+                encoded_output = base64.b64encode(output_data.encode("utf-8")).decode(
+                    "utf-8"
+                )
+
+                output_record = {
+                    "recordId": record["recordId"],
+                    "result": "Ok",
+                    "data": encoded_output,
+                }
+                output_records.append(output_record)
+            else:
+                output_record = {
+                    "recordId": record["recordId"],
+                    "result": "Dropped",
+                    "data": record["data"],
+                }
+                output_records.append(output_record)
+            logger.info(f"Processed record with {len(processed_logs)} logs")
+    except Exception as e:
+        logger.error(f"Error processing logs: {str(e)}")
+    return {"records": output_records}
+
+
+def make_prefixes():
+    environment = os.getenv("ENVIRONMENT")
+    if not environment:
+        raise RuntimeError("environment is required")
+
+    rds_prefix = "cg-aws-broker-"
+    environment_suffixes = {
+        "production": "prod",
+        "staging": "stage",
+        "development": "dev",
+    }
+    if environment not in environment_suffixes:
+        raise RuntimeError(f"environment is invalid: {environment}")
+
+    rds_prefix += environment_suffixes[environment]
+
+    return rds_prefix
+
+
+def process_logs(logs, client, region, account_id, rds_prefix):
+    try:
+        return_logs = []
+        resource_name = logs["logGroup"].split("/")[4]
+
+        tags = get_resource_tags_from_log(
+            resource_name, client, region, account_id, rds_prefix
+        )
+
+        if len(tags.keys()) > 0:
+            for event in logs["logEvents"]:
+                entry = {
+                    "logGroup": logs["logGroup"],
+                    "logStream": logs["logStream"],
+                    "message": event["message"],
+                    "timestamp": event["timestamp"],
+                    "Tags": tags,
+                }
+                return_logs.append(entry)
+        else:
+            return None
+        return return_logs
+
+    except Exception as e:
+        logger.error(f"Could not process logs: {e}")
+        return None
+
+
+def get_resource_tags_from_log(
+    resource_name, client, region, account_id, rds_prefix
+) -> dict:
+    tags = {}
+    try:
+        if resource_name is not None and resource_name.startswith(rds_prefix):
+            arn = f"arn:aws-us-gov:rds:{region}:{account_id}:db:{resource_name}"
+            tags = get_tags_from_arn(arn, client)
+    except Exception as e:
+        logger.error(f"Error with getting tags for resource: {e}")
+    return tags
+
+
+@lru_cache(maxsize=256)
+def get_tags_from_arn(arn, client) -> dict:
```

Comment:
Seems like we should just be able to use the function already defined in the other Lambda file here? Maybe just move that function into another file and import it into both Lambdas?

Reply:
we'll have to change up the terraform a bit but i don't see why not

Reply:
Why would the Terraform need to change? It still just references the filename containing the Lambda handler and the function, like `transform_lambda.lambda_handler`

Reply:
So in the terraform we pass the file from a github raw tag so it reads in just the file. We do this so that the lambda stays very lightweight and lowers the overhead of having to support test libraries also and sizing. The smaller what we feed in the less it has to load in everytime it runs.

So the terraform reads in the one file and zips it up. we can change it so the terraform reads in multiple files

Reply:
we might be able to get away with a lambda layer to make it quickeras both need pretty much the same things

Reply:
Alright. I would say ignore this feedback then. It's not worth complicating the Terraform just for this minor benefit